src/ai.js

```diff
@@ -1,16 +1,651 @@
+            let result = simulation.interact.doTurnLogic(index.toString());
+            this.on_card = null;
+
+            if (result !== true && !result instanceof Card || typeof result === "string") return; // Invalid move
+
+            let score = this._evaluate(simulation);
+
+            if (score <= best_move[1]) return;
+
+            // This card is now the best move
+            best_move = [card, score];
+        });
+
+        this._restoreGame();
+
+        let score = best_move[1];
+        best_move = best_move[0];
+
+        if (!best_move) {
+            // Couldn't play any cards
+            if (this.canAttack) best_move = "attack";
+
+            // TODO: Add hero power
+            // TODO: Add use locations
+
+            else best_move = "end";
+
+            this.history.push(["chooseMove", best_move]);
+        } else {
+            this.history.push(["chooseMove", [best_move.name, score]]);
+
+            best_move = this.plr.hand.indexOf(best_move) + 1;
+        }
+
+        if (best_move == "end") {
+            this.canAttack = true;
+        }
+
+        return best_move;
+    }
+
+    /**
+     * 
+     * @param {Card | Player} attacker 
+     * @param {Card | Player} target 
+     * @param {[Card | Player, Card | Player, number]} best_attack 
+     * @param {Game} simulation
+     * 
+     * @returns {[Card | Player, Card | Player, number]}
+     */
+    _attackTarget(attacker, target, best_attack, simulation) {
+        let health = target.getHealth();
+        let abck, tbck;
+
+        if (attacker.classType == "Card") abck = attacker.createBackup();
+        if (target.classType == "Card") tbck = target.createBackup();
+        
+        let result = simulation.attack(attacker, target);
+        if (result !== true) return best_attack; // Invalid attack
+
+        let score = this._evaluate(simulation);
+
+        if (attacker.classType == "Card") {
+            attacker.restoreBackup(attacker.backups[abck]);
+            attacker.ready();
+        }
+
+        if (target.classType == "Card") target.restoreBackup(target.backups[tbck]);
+        else target.health = health;
+
+        if (score <= best_attack[2]) return best_attack;
+
+        return [attacker, target, score];
+    }
+
+    /**
+     * Makes the ai attack
+     *
+     * @returns {[Card | Player | -1 | null, Card | Player | -1 | null]} Attacker, Target
+     */
+    attack() {
+        // FIXME: The ai doesn't attack.
+        let simulation = this._createSimulation();
+
+        let board = game.board;
+        let thisboard = board[this.plr.id];
+        let opboard = board[this.plr.getOpponent().id];
+
+        /**
+         * @type {[Card | Player, Card | Player, number]}
+         */
+        let best_attack = [];
+
+        [...thisboard, this.plr].forEach(attacker => {
+            if (attacker.attackTimes <= 0 || attacker.sleepy || attacker.dormant || attacker.attack || (attacker.getAttack && !attacker.getAttack())) return;
```

Comment:
Ew wtf. Put this in a function maybe. Doesn't the sent ai model have something simular? If so, use that instead.